_posts/2021-07-07-cockpit-248.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Cockpit 248
+author: garrett
+date: '2021-07-07'
+tags: cockpit, machines, ostree, grafana, pcp, redis, metrics, pam
+slug: cockpit-248
+category: release
+summary: 'Metrics, PAM, Machines, OSTree, and git main branch changes'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 248, Cockpit-Podman 32, Cockpit-Machines 247, and Cockpit-OSTree 187:
+
+
+## Metrics: Install missing packages
+
+[Cockpit 247](https://cockpit-project.org/blog/cockpit-247.html) introduced setting up a machine for Grafana metrics. Yet, this only worked if the `pcp` and `redis` packages were already installed. In Cockpit 248 and after, selecting these options will install any missing packages.
+
+![screenshot of install missing packages]({{ site.baseurl }}/images/248-metrics-install-missing-packages.png)
+
+![screenshot of install missing packages]({{ site.baseurl }}/images/248-metrics-install-missing-packages-2.png)
+
+This completes the story of [setting up PCP and Grafana metrics with Cockpit](https://cockpit-project.org/blog/pcp-grafana.html).
+
+## PAM: Deprecate `pam_cockpit_cert` module
+
+The `pam_cockpit_cert` module is deprecated and no longer required for client certificate authentication. Cockpit continues to ship a stub module which returns `PAM_IGNORE`, but a future release will drop this stub.
+
+Upgrading packages will not update a modified `/etc/pam.d/cockpit` file. If you have modified this file, please take care to remove the relevant entry.
+
+## Machines: Bug fixes and improvements
+
+Cockpit-Machines has a new release with several bug fixes and minor improvements.
+
+## OSTree: Bump epoch to 1
+
+The cockpit-ostree RPM package is being released as version `1:186`, with a newly introduced epoch. This new version will re-enable upgrading from an unintentional "version 999" of cockpit-ostree. Version 999 was accidentally published while testing the release infrastructure. These package changes only affect ostree-based distributions, such as Fedora CoreOS.
+
+## Developers: `main` branch switch and build system updates
+
+Over the past few months, many Cockpit sub-projects have switched git branches to `main`. As of this release, all Cockpit repositories&mdash;including the main Cockpit repository&mdash;have finished the switch. Developers with checked out git repositories should update their branches.
```

Comment:
I'd make it more explicit: "changed to using main as their primary development branch" or "changed the name of their default branch to main" or so.  "switched git branches to main" is annoyingly unspecific.


"Update their branches" → maybe give a link to the page about this (see topic in #cockpit IRC channel)